### PR TITLE
POLIO-1461: The analytics script is not executed on the embedded calendar

### PR DESCRIPTION
**Pull request by hakifran**
Explain what problem this PR is resolving
- The analytics script is not executed on the embedded calendar
- This PR is about to fix it
Related JIRA tickets : [POLIO-1461](https://bluesquare.atlassian.net/browse/POLIO-1461)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes
- Get all analytics from all accounts
- Add the analytics script to django template
## How to test
-Go on the embedded calendar and check if the analytics script is running
## Print screen / video
[Screencast from 2024-01-31 14-59-50.webm](https://github.com/BLSQ/iaso/assets/19631540/f1752d8b-963d-45ec-a282-786040fe2f6b)

[POLIO-1461]: https://bluesquare.atlassian.net/browse/POLIO-1461?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: hat/templates/iaso/base.html [hunk 1]
```diff
@@ -26,6 +26,11 @@
   <meta property="og:image" content="" />
   <meta property="og:url" content="{{ request.build_absolute_uri }} " />
   {% endif %}
+  {% if ANALYTICS_SCRIPTS %}
+    {% for analytic_script in ANALYTICS_SCRIPTS %}
```
Comment (madewulf) [inline hat/templates/iaso/base.html:30]: You can't have all the analytics script of all accounts displayed here, that defeats the point of having a different script per account

File: hat/dashboard/views.py [hunk 2]
```diff
@@ -35,7 +41,10 @@ def iaso(request: HttpRequest) -> HttpResponse:
 @require_http_methods(["GET"])
 def embeddable_iaso(request: HttpRequest) -> HttpResponse:
     """Embeddable iaso page without login requirement and with correct header"""
-    response = _base_iaso(request)
+    all_analytics_scripts = set(
+        [account.analytics_script for account in Account.objects.all() if account.analytics_script]
```
Comment (mathvdh) [inline hat/dashboard/views.py:45]: It's weird to have ALL the analytics_script from all the Accounts loaded ? Or I don't understand somelthing

**Review by mathvdh — CHANGES_REQUESTED** [review]
See my comment, otherwise I think it's good.

File: hat/templates/iaso/base.html (hunk 1 above)
Comment (mathvdh) [reply hat/templates/iaso/base.html:30]: Ah this seems to be about the same thing I'm commenting below

Comment (hakifran) [reply hat/templates/iaso/base.html:30]: As you can see the analytics script will be added only for the embedded calendar, which is accessible by not authenticated user. And on the embedded calendar there can be displayed campaigns from different accounts, that's why I tried to get all the analytics script from all accounts. I don't know if it's clear or if you want we can have a 5min call. thanks

**Review by madewulf — APPROVED** [review]

**Pull request merged**